react/features/base/tracks/actions.js

```diff
@@ -397,19 +397,29 @@ export function trackAdded(track) {
     return async (dispatch, getState) => {
         track.on(
             JitsiTrackEvents.TRACK_MUTE_CHANGED,
-        () => {
-            if (getSourceNameSignalingFeatureFlag(getState()) && track.getVideoType() === VIDEO_TYPE.DESKTOP) {
-                dispatch(screenshareTrackMutedChanged(track));
+            () => {
+                dispatch(trackMutedChanged(track));
+
+                if (getMultipleVideoSupportFeatureFlag(getState()) && track.getVideoType() === VIDEO_TYPE.DESKTOP) {
+                    dispatch(screenshareTrackMutedChanged(track));
+                }
             }
-            dispatch(trackMutedChanged(track));
-        });
+        );
         track.on(
             JitsiTrackEvents.TRACK_VIDEOTYPE_CHANGED,
-            type => dispatch(trackVideoTypeChanged(track, type)));
+            type => {
+                dispatch(trackVideoTypeChanged(track, type));
+
+                if (getMultipleVideoSupportFeatureFlag(getState())) {
```

Comment:
Can we not dispatch a new `screenshareVideoTypeChanged` action here and instead add multi-stream handling in /base/tracks/middleware for `TRACK_UPDATED` action type ? Also `screenshareVideoTypeChanged` doesn't sound right since it is a single track whose videoType can either be camera or desktop.

Reply:
I'll look into seeing if we can just use `TRACK_UPDATED`. If we go with this approach we'll need to do a comparison between the previous and new track to see if the mute/unmute and camera/desktop states changed